Tests/Client/Custom Assertions/AssertNetworkRequest.swift

```diff
@@ -0,0 +1,150 @@
+//
+//  AssertNetworkRequest.swift
+//  StreamChatClientTests
+//
+//  Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import XCTest
+@testable import StreamChatClient
+
+
+/// Synchronously waits for a network request to be made and asserts its properties.
+///
+/// The function always uses the latest request `RequestRecorderURLProtocol` records. If no request has
+/// been made within the `timeout` period, this assertion fails with the time-out error.
+///
+/// The values specified in the `headers`, `queryParameters` and `body` represents the mandatory subset of
+/// the values the request must have. A request is valid even when it contains additional parameters than
+/// the ones specified in these values.
+///
+/// - Parameters:
+///   - method: The HTTP method the request.
+///   - path: The `path` part of the request's URL.
+///   - headers: The headers required for the request.
+///   - queryParameters: The query parameters required for the request.
+///   - body: The expected body of the request.
+///   - timeout: The maximum time this function waits for a request to be made.
+///
+func AssertNetworkRequest(
+    method: Endpoint.Method,
+    path: String,
+    headers: [String: String]?,
+    queryParameters: [String: String]?,
+    body: [String: Any]?,
+    timeout: TimeInterval = 0.5,
+    file: StaticString = #file,
+    line: UInt = #line
+) {
```

Comment:
We don't start parameter from a new line.
```suggestion
func AssertNetworkRequest(method: Endpoint.Method,
                          path: String,
                          headers: [String: String]?,
                          queryParameters: [String: String]?,
                          body: [String: Any]?,
                          timeout: TimeInterval = 0.5,
                          file: StaticString = #file,
                          line: UInt = #line) {
```

Reply:
fixed